### PR TITLE
Consolidate apigateways

### PR DESCRIPTION
**Pull request by colin-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/grapl/issues/438

### What changes does this PR make to Grapl? Why?
This PR consolidates our API Gateways into one gateway. There are therefor a number of changes

1) The grapl-cdk defines one top level 'EdgeApi' gateway
2) This gateway has one single rate limit policy
3) Path based routing where the base of the path is the responsibility that maps to the lambda
4) Some minor code changes to handle these path changes

### How were these changes tested?
Fresh deployment to AWS.

### CODE REVIEW

File: src/js/grapl-cdk/bin/deployment_parameters.ts [hunk 1]
```diff
@@ -2,10 +2,11 @@ import * as ec2 from '@aws-cdk/aws-ec2';
 
 module HardcodedDeploymentParameters {
     // ex: 'Grapl-my-deployment'
+
     export const deployName = undefined;
 
     // defaults to 'latest'
-    export const graplVersion = undefined;
+    export const graplVersion = 'latest';
```
Comment (jgrillo-grapl) [inline src/js/grapl-cdk/bin/deployment_parameters.ts:9]: This should remain `undefined`. If it doesn't indeed default to latest below as the docstring indicates then we should fix that.

File: src/js/grapl-cdk/edge_ux/index.html [hunk 2]
```diff
@@ -1 +1 @@
...
...turn e}var n={},o={1:0},u=[];function l(r){if(n[r])return n[r].exports;var t=n[r]={i:r,l:!1,exports:{}};return e[r].call(t.exports,t,t.exports,l),t.l=!0,t.exports}l.m=e,l.c=n,l.d=function(e,r,t){l.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},l.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},l.t=function(e,r){if(1&r&&(e=l(e)),8&r)return e;if(4&r&&"object"==typeof e&&e&&e.__esModule)return e;var t=Object.create(null);if(l.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:e}),2&r&&"string"!=typeof e)for(var n in e)l.d(t,n,function(r){return e[r]}.bind(null,n));return t},l.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return l.d(r,"a",r),r},l.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},l.p="/";var f=this["webpackJsonpgrapl-fe"]=this["webpackJsonpgrapl-fe"]||[],a=f.push.bind(f);f.push=r,f=f.slice();for(var p=0;p<f.length;p++)r(f[p]);var i=a;t()}([])</script><script src="/static/js/2.924c1906.chunk.js"></script><script src="/static/js/main.b63c5b67.chunk.js"></script></body></html>
```
Comment (jgrillo-grapl) [inline src/js/grapl-cdk/edge_ux/index.html:1]: It looks like a lot of compilation artifacts got committed? Most of the files in this PR appear to be minified js.

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts [hunk 3]
```diff
@@ -709,76 +710,10 @@ export class ModelPluginDeployer extends cdk.NestedStack {
...
-            },
-            throttle: {
-                // per minute
-                rateLimit: 50,
-                burstLimit: 50,
-            },
-        });
-
-        if (props.watchful) {
-            props.watchful.watchApiGateway(
```
Comment (jgrillo-grapl) [inline src/js/grapl-cdk/lib/grapl-cdk-stack.ts:731]: Did we lose per-path HTTP metrics in this PR? If so, how can we get them back?

**Review by jgrillo-grapl — COMMENTED** [review]
A few questions. Mostly confused by the amount of generated code this PR introduces, and wondering if we've lost granularity in our HTTP metrics. I don't really know what I'm looking at for frontend stuff so these comments are non-blocking.

File: src/js/grapl-cdk/edge_ux/index.html (hunk 2 above)
Comment (colin-grapl) [reply src/js/grapl-cdk/edge_ux/index.html:1]: I think we do commit these so that the localgrapl demo doesn't require a separate build step. I'm not sure.

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts (hunk 3 above)
Comment (colin-grapl) [reply src/js/grapl-cdk/lib/grapl-cdk-stack.ts:731]: I'm actually not sure. I would sort of assume that watchful just does that, but I'll check. If not, I can try to take a quick pass to fix this.

Comment (colin-grapl) [reply src/js/grapl-cdk/lib/grapl-cdk-stack.ts:731]: Yeah, we did regress here, fixing now.

File: src/python/engagement_edge/src/engagement_edge.py [hunk 4]
```diff
@@ -36,6 +36,8 @@
 LOGGER.setLevel(GRAPL_LOG_LEVEL)
 LOGGER.addHandler(logging.StreamHandler(stream=sys.stdout))
 
+print("abc 123")
```
Comment (inickles-grapl) [inline src/python/engagement_edge/src/engagement_edge.py:39]: testing/debug code?

Comment (inickles-grapl) [reply src/python/engagement_edge/src/engagement_edge.py:39]: should probably remove.

**Review by inickles-grapl — APPROVED** [review]

**Review by jgrillo-grapl — APPROVED** [review]

File: src/rust/node-identifier/tests/assetdb_integration_tests.rs [hunk 5]
```diff
@@ -1,35 +0,0 @@
-use grapl_graph_descriptions::graph_description::host::*;
```
Comment (jgrillo-grapl) [inline src/rust/node-identifier/tests/assetdb_integration_tests.rs:1]: why was this deleted?

File: src/rust/node-identifier/tests/sessiondb_integration_tests.rs [hunk 6]
```diff
@@ -1,309 +0,0 @@
-use std::time::Duration;
```
Comment (jgrillo-grapl) [inline src/rust/node-identifier/tests/sessiondb_integration_tests.rs:1]: And why was this deleted?
